### ISSUE TITLE
force:source:retrieve -x package.xml with mix of wildcard / named entities broken (e.g. CustomObject)

### ISSUE THREAD

**Issue by svenfinitiv** [forcedotcom/cli#1205]
### Summary

Starting with 7.118.0 we noticed that standard objects, standard field and standard value sets are no longer retrieved when combining the wildcard * with named entities. Version 7.117.0 is not affected.

### Steps To Reproduce:

Use this package.xml (it doesn't matter how the members for CustomObject are combined)

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
    <types>
        <members>Account</members>
        <name>CustomObject</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomObject</name>
    </types>
    <version>53.0</version>
</Package>

```
```bash
sfdx force:source:retrieve -x package.xml
```

### Expected result

Changes to the Account object are retrieved (this also affected profiles and permission sets).

### Actual result

Only custom objects are retrieved, Account is unchanged.

### System Information

Run `sfdx version --verbose --json` and paste the output here.

```json
{
	"cliVersion": "sfdx-cli/7.119.2",
	"architecture": "darwin-x64",
	"nodeVersion": "node-v14.17.6",
	"pluginVersions": [
		"@oclif/plugin-autocomplete 0.3.0 (core)",
		"@oclif/plugin-commands 1.3.0 (core)",
		"@oclif/plugin-help 3.2.3 (core)",
		"@oclif/plugin-not-found 1.2.4 (core)",
		"@oclif/plugin-plugins 1.10.1 (core)",
		"@oclif/plugin-update 1.5.0 (core)",
		"@oclif/plugin-warn-if-update-available 1.7.0 (core)",
		"@oclif/plugin-which 1.0.3 (core)",
		"@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)",
		"alias 1.1.10 (core)",
		"apex 0.2.8 (core)",
		"auth 1.7.1 (core)",
		"config 1.2.35 (core)",
		"custom-metadata 1.0.12 (core)",
		"data 0.6.1 (core)",
		"generator 1.2.0 (core)",
		"limits 1.2.1 (core)",
		"org 1.7.0 (core)",
		"salesforce-alm 52.3.5 (core)",
		"schema 1.0.8 (core)",
		"sfdx-cli 7.119.2 (core)",
		"source 1.1.0 (core)",
		"telemetry 1.2.4 (core)",
		"templates 52.1.0 (core)",
		"trust 1.0.8 (core)",
		"user 1.4.0 (core)"
	],
	"osVersion": "Darwin 20.6.0"
}
```

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-9960907](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000BPQYuYAP/view)

**Comment by iowillhoit** [comment]
Was able to reproduce this, got a work item created. Thanks for submitting this issue.

**Comment by cristiand391** [comment]
This was fixed in [sdr@4.5.2](https://github.com/forcedotcom/source-deploy-retrieve/blob/main/CHANGELOG.md#452-2021-10-03), the fix is available in sfdx latest-rc, which will be promoted next Thursday.
